spec/services/document_unpublishing_service_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+RSpec.describe DocumentUnpublishingService do
+  describe "#retire" do
+    let(:document) { create(:document) }
+    let(:explanatory_note) { "The document is out of date" }
+
+    it "withdraws a document in publishing-api with an explanatory note" do
+      stub_publishing_api_unpublish(document.content_id, body: { type: "withdrawal", explanation: explanatory_note, locale: "en" })
```

Comment:
On these stubs here it'd be good to use the locale from the document too (I appreciate it may be news to you that we even store it) save the tests breaking if we set different locales in the factory